libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c

```diff
@@ -336,7 +362,28 @@ static void prvRxSelectSet( ss_ctx_t * ctx,
 
 static void prvRxSelectClear( ss_ctx_t * ctx )
 {
-    /* TODO */
+    /* Inform the vTaskRxSelect to delete itself. */
+    xEventGroupSetBits( ctx->rx_EventGroup, SOCKETS_START_DELETION );
+
+    /* Wait for the task to delete itself. */
+    if( xEventGroupWaitBits( ctx->rx_EventGroup, SOCKETS_COMPLETE_DELETION, pdTRUE, pdTRUE, portMAX_DELAY ) == SOCKETS_COMPLETE_DELETION )
+    {
+        /* The task 'vTaskRxSelect' has started deleting itself. */
+    }
+    else
+    {
+        /* Control should never reach here as this task should wait on the event group bits indefinately. */
+        configASSERT( pdTRUE );
+    }
```

Comment:
I think this is an assumption.  I think indefinite blocking requires INCLUDE_vTaskSuspend.  If a customer configures that to be 0, then the function could return due to timeout.  Perhaps wrap this in a while loop?

Reply:
Ah yes! I overlooked that. I have changed that to a while. Thanks for catching that Dan.